FastSimulation/Tracking/plugins/FastTrackingMaskProducer.cc

```diff
@@ -0,0 +1,175 @@
+#include "FWCore/ParameterSet/interface/ParameterSet.h"
+#include "FWCore/Framework/interface/ESHandle.h"
+#include "DataFormats/Common/interface/Handle.h"
+#include "DataFormats/Common/interface/OwnVector.h"
+#include "DataFormats/TrackerRecHit2D/interface/SiTrackerGSRecHit2DCollection.h"
+#include "DataFormats/TrackerRecHit2D/interface/SiTrackerGSMatchedRecHit2DCollection.h"
+#include "DataFormats/TrackerRecHit2D/interface/SiTrackerGSRecHit2D.h"
+#include "DataFormats/TrackerRecHit2D/interface/SiTrackerGSMatchedRecHit2D.h"
+#include "DataFormats/TrackReco/interface/TrackFwd.h"
+#include "FastSimulation/Tracking/plugins/FastTrackingMaskProducer.h"
+#include <vector>
+#include <stdio.h>
+
+FastTrackingMaskProducer::FastTrackingMaskProducer(const edm::ParameterSet& conf) 
+  : oldHitMasks_exists_(false)
+  , oldHitCombinationMasks_exists_(false)
+  , overRideTrkQuals_(false)
+  , filterTracks_(false)
+{
+  // Main products                                                                                                                 
+  produces<std::vector<bool> >("hitMasks");
+  produces<std::vector<bool> >("hitCombinationMasks");
+
+  // Track collection                                                                                                                   
+  edm::InputTag trackCollectionTag = conf.getParameter<edm::InputTag>("trackCollection");
+  trackToken_ = consumes<reco::TrackCollection>(trackCollectionTag);
+
+  // old hit masks
+  oldHitMasks_exists_ = conf.exists("oldHitMasks");
+  if (oldHitMasks_exists_){
+    edm::InputTag hitMasksTag = conf.getParameter<edm::InputTag>("oldHitMasks");
+    hitMasksToken_ = consumes<std::vector<bool> >(hitMasksTag);
+  }
+
+  // old hit combination masks
+  oldHitCombinationMasks_exists_ = conf.exists("oldHitCombinationMasks");
+  if (  oldHitCombinationMasks_exists_){
+    edm::InputTag hitCombinationMasksTag = conf.getParameter<edm::InputTag>("oldHitCombinationMasks");
+    hitCombinationMasksToken_ = consumes<std::vector<bool> >(hitCombinationMasksTag);
+  }
+  
+  // read track quality from value map rather than from track itself
+  overRideTrkQuals_ = conf.exists("overrideTrkQuals");
+  if(  overRideTrkQuals_ ){
+    edm::InputTag trkQualsTag = conf.getParameter<edm::InputTag>("overrideTrkQuals");
+    if(trkQualsTag == edm::InputTag(""))
+      overRideTrkQuals_ = false;
+    else
+      trkQualsToken_ = consumes<edm::ValueMap<int> >(trkQualsTag);
+  }
+
+  // required track quality
+  trackQuality_=reco::TrackBase::undefQuality;
+  if (conf.exists("TrackQuality")){
+    filterTracks_=true;
+    std::string trackQualityStr = conf.getParameter<std::string>("TrackQuality");
+    if ( !trackQualityStr.empty() ) {
+      trackQuality_=reco::TrackBase::qualityByName(trackQualityStr);
+    }
+  }
+}
+
+void
+FastTrackingMaskProducer::produce(edm::Event& e, const edm::EventSetup& es)
+{
+  // the products
+  std::auto_ptr<std::vector<bool> > hitMasks(new std::vector<bool>());
```

Comment:
@lveldere , @kripav What about writing out a ValueMap instead? Also, you know the size of the vector already. Just initialize it directly